pkg/ipamd/ipamd.go

```diff
@@ -187,6 +188,38 @@ type IPAMContext struct {
 	terminating            int32 // Flag to warn that the pod is about to shut down.
 }
 
+// UnmanagedENISet keeps a set of ENI IDs for ENIs tagged with "node.k8s.amazonaws.com/no_manage"
+type UnmanagedENISet struct {
+	set  map[string]bool
+	lock sync.RWMutex
```

Comment:
note that you can "inherit" from `sync.RWMutex` here instead of making a named `lock` field:

```go
type UnmanagedENISet struct {
    sync.RWMutex
    enis map[string]bool
}
```

and then you can call `Lock()` directly, potentially from a mutator method of `*UnamanagedENISet`) like so:

```go
func (u *UnmanagedENISet) add(eniID string) {
    u.Lock()
    defer u.Unlock()
    if len(u.data) == 0 {
        u.data = make(map[string]bool)
    }
    u.data[eniID] = true
}
```

Reply:
Much nicer, thanks!